### DISCUSSION TITLE
Schema with 1:1 relation not updateable

### DISCUSSION THREAD

**Discussion by LewisMorgans** [prisma/prisma#10603] [Q&A] [answered]
So for some reason I can't seem to update a user. My logic before the prisma call works fine, updating the returned user with the new values. Which in this case is just a single property in the options object related to the user. But when trying to call prisma.update it goes hay wire.

### **Error Message**

```
Unknown arg `typeScript` in data.options.typeScript for type OptionsUpdateOneWithoutUserInput. Did you mean `upsert`? Available args:
type OptionsUpdateOneWithoutUserInput {
  create?: OptionsCreateWithoutUserInput | OptionsUncheckedCreateWithoutUserInput
  connectOrCreate?: OptionsCreateOrConnectWithoutUserInput
  upsert?: OptionsUpsertWithoutUserInput
  disconnect?: Boolean
  delete?: Boolean
  connect?: OptionsWhereUniqueInput
  update?: OptionsUpdateWithoutUserInput | OptionsUncheckedUpdateWithoutUserInput
}
```

### **Schema**

```
model User {
    id       String @id @default(dbgenerated()) @map("_id") @db.ObjectId
    email    String
    password String
    role     String
    options Options?
}

model Options {
    user   User   @relation(fields: [userId], references: [id])
    userId String @unique @id @map("_id") @db.ObjectId 
    typeScript Boolean
    rxjs Boolean
 }
```

### **Method**
```
  public async updateDetails(x): Promise<User | void> {
    const user: User = await this.getUser('61ae8d8085e3a2a5b4a6bd12');
    let options: Options;
    const details = {
      typeScript: false,
    };

    for (const [key, value] of Object.entries(details)) {
      options = { ...user.options, [key]: value };
      delete user?.id;
    }
    const updatedUserRecord: User = {
      ...user,
      options,
    };

    console.log(updatedUserRecord);

    return await prisma.user.update({
      where: {
        id: '61ae8d8085e3a2a5b4a6bd12',
      },
      data: { ...updatedUserRecord },
    });
  }
```

Going to tear my hair out soon. Would appreciate some help here.

A user has a collection of options. I'm trying to update the entire user object, including the options nested object. 

![image](https://user-images.githubusercontent.com/28410957/144942609-8ca7d830-b901-4793-8a89-41f917dd5e65.png)

**Comment by chemicalkosek** [comment, marked as answer]
You're probably looking for something like this. See if it works:

```javascript
  public async updateDetails(x): Promise<User | void> {
    const user: User = await this.getUser('61ae8d8085e3a2a5b4a6bd12');
    let options: Options;
    const details = {
      typeScript: false,
    };

    const updatedUser = {...user};
    delete updatedUser.id;
    delete updatedUser.options;
    options = {...user.options, ...details}

    return await prisma.user.update({
      where: {
        id: '61ae8d8085e3a2a5b4a6bd12',
      },
      data: { 
        ...updatedUser,
        options: {
          update: {
            ...options
          }
        }           
        },
    });
  }
```

**Reply by LewisMorgans** [reply]
Ah brilliant, thank you for that. That indeed worked :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
